### PR TITLE
docs: Add configurable gradient clipping example to custom trainer guide

### PR DESCRIPTION
**Pull request by artest08**
Documents how to override optimizer_step() to make max_norm configurable, with RT-DETR-flavored example and reference values for common detection codebases.

Adds a `## Configurable Gradient Clipping` section to the custom trainer guide showing how to subclass `RTDETRTrainer` (or `DetectionTrainer`) and override `optimizer_step()` to make `max_norm` configurable. The default trainer hardcodes `max_norm=10.0`, which is loose for DETR-family detectors that typically use `0.1`; the new section gives users a copy-paste recipe to tighten the clip without touching `BaseTrainer` or `default.yaml`.

```python
import torch

from ultralytics import RTDETR
from ultralytics.models.rtdetr.train import RTDETRTrainer

class CustomClipTrainer(RTDETRTrainer):
    """RT-DETR trainer with configurable gradient clipping."""

    clip_grad_norm = 0.1

    def optimizer_step(self):
        self.scaler.unscale_(self.optimizer)
        if self.clip_grad_norm > 0:
            torch.nn.utils.clip_grad_norm_(self.model.parameters(), max_norm=self.clip_grad_norm)
        self.scaler.step(self.optimizer)
        self.scaler.update()
        self.optimizer.zero_grad()
        if self.ema:
            self.ema.update(self.model)

RTDETR("rtdetr-l.pt").train(data="coco8.yaml", epochs=20, trainer=CustomClipTrainer)
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📝 This PR expands the custom trainer guide by adding a new section on configurable gradient clipping, helping users better stabilize training for different model families.

### 📊 Key Changes
- Added **gradient clipping** to the list of supported custom trainer examples, increasing the guide from **six to seven** customization patterns.
- Updated the page metadata and keywords to include **gradient clipping** for better documentation discoverability.
- Introduced a new **“Configurable Gradient Clipping”** section in [`custom-trainer.md`](https://github.com/ultralytics/ultralytics).
- Added a practical example showing how to subclass `RTDETRTrainer` and override `optimizer_step()` to set a custom `clip_grad_norm` value.
- Documented recommended clipping behavior for different architectures:
  - **RT-DETR / DETR-style models:** tighter clipping like `0.1`
  - **YOLO models:** default loose clipping like `10.0`
  - **Disable clipping:** set to `0`
- Clarified that the same approach also works for YOLO trainers by switching the base trainer class, with no changes needed to the optimizer step logic.

### 🎯 Purpose & Impact
- 🎯 Helps users understand **when and why to tune gradient clipping**, especially for transformer-based detectors like RT-DETR, DEIM, and DINO.
- 🚀 Improves training stability for models that are more sensitive to gradient spikes, reducing trial-and-error during custom training.
- 📚 Makes the custom trainer docs more complete and practical for advanced users building tailored training workflows.
- 🤝 Benefits both experts and newer users by giving a clear, copy-ready example instead of requiring them to inspect trainer internals manually.
- 🔧 Encourages safer customization of Ultralytics training behavior without changing core library code.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @artest08, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with a smooth review process, and an engineer will assist you soon 😊 Please review the checklist below:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The documentation update looks clean. The new gradient clipping example matches the current trainer flow, and I didn't find any actionable correctness or maintainability issues in this diff.

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉 Thank you @artest08, with contributions from @Laughing-q, for strengthening the custom trainer docs with a clear and practical guide to configurable gradient clipping.

As Leonardo da Vinci said, *“Simplicity is the ultimate sophistication.”* This update reflects that well: a small, focused addition that gives users a safer and more intuitive way to stabilize training across RT-DETR, DEIM, DINO, and YOLO workflows without touching core code.

Really appreciate the thoughtful documentation improvements, copy-ready example, and the extra clarity for both advanced users and newcomers. Great contribution to making Ultralytics training customization more accessible and reliable.
